cloudinit/config/cc_write_files.py

```diff
@@ -86,8 +88,23 @@ def write_files(name, files, owner: str):
             )
             continue
         path = os.path.abspath(path)
-        extractions = canonicalize_extraction(f_info.get("encoding"))
-        contents = extract_contents(f_info.get("content", ""), extractions)
+        # Read content from provided URL, if any, or decode from inline
+        contents = read_url_or_decode(
+            f_info.get("source", None),
+            ssl_details,
+            f_info.get("content", None),
+            f_info.get("encoding", None),
+        )
+        if contents == None:
+            LOG.warning(
```

Comment:
Sorry, my last comment wasn't clear, but I don't think this is what we want either.

If we have a file source, we're expecting to pull in some specific data, and if that fails, we shouldn't be leaving empty files around.

However,  `content` can be empty and if so, we should expect to write an empty file. E.g.,
```
#cloud-config
write_files:
  - path: /tmp/thing
```

With that, we'd expect `/tmp/thing` to be written and empty. As implemented here, this no longer works. So to summarize:
* If source and content, attempt source but fallback to content
* If source and no content, attempt source but do not create file if we fail to retrieve it
* If content and no source, write the content
* If no source and no content, write an empty file

Reply:
Ah, I somehow thought I'd handled that. Fixed now, and there's a test for this behavior as well.